front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/[dsId]/tokenize.ts

```diff
@@ -0,0 +1,121 @@
+import type { CoreAPITokenType, WithAPIErrorResponse } from "@dust-tt/types";
+import { CoreAPI } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { withPublicAPIAuthentication } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import {VaultResource} from "@app/lib/resources/vault_resource";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+export type PostDatasourceTokenizeBody = {
+  text: string;
+};
+
+const PostDatasourceTokenizeBodySchema = t.type({
+  text: t.string,
+});
+
+type PostDatasourceTokenizeResponseBody = {
+  tokens: CoreAPITokenType[];
+};
+
+/**
+ * @ignoreswagger
+ * This endpoint is not to be included in the public API docs.
+ */
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<PostDatasourceTokenizeResponseBody>
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  // Handling the case where vId is undefined to keep support for the legacy endpoint (not under
+  // vault, global vault assumed for the auth (the authenticator associated with the app, not the
+  // user)).
+  let { vId } = req.query;
+  if (typeof vId !== "string") {
+    vId = (await VaultResource.fetchWorkspaceGlobalVault(auth)).sId;
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
+    { origin: "v1_data_sources_tokenize" }
+  );
+  if (!dataSource || dataSource.vault.sId !== vId) {
```

Comment:
Same issue here - legacy endpoint won't work for managed ds as it assumes global vault and managed ds are in system vault - all legacy endpoints used by connectors will break ?

Reply:
Yes realised that while commuting back. Will adress tomorrow.

There's a bit more work to allow searching connections as well.